### PR TITLE
D-14518 

### PR DESCRIPTION
**Pull request by walkerrandolphsmith**
There are several changes so I'd like to get feedback sooner in the process.

---

Rotate component is being removed altogether in favor of Icons handling a rotate prop in order to reduce the DOM depth.

---

SingleSelect can receive a set of `items` such that each item must have at least a `label` and a `value`.

1. Given a user types in the field and no options match, then tabbing and enter keypress clear the field and no selection is made.

2. Given a user types in the field and an option partially matches, then tabbing and enter keypress fill out the field with the remaining text from the matched option and a selection is made.

3. Typing into the field can filter the options. The consumer can determine the filtering behavior.

4. Arrow keys navigation the options when they exist and the field is active.

5. Escape keypress dismisses the options and blurs the field.

6. The options can visually indicate the previous selection.

### CODE REVIEW

**Comment by walkerrandolphsmith** [comment]
If and when we migrate to font icons then the rotation can be applied via a class.

**Comment by jallers** [comment]
Code approach and API looks good.

Found a few issues.

Filtered item did not appear anchored to input selection.
image

Clearing the text in the input should stop highlighting the item in the pop-up list
image

Alt+Down and Alt+Up should show and hide dropdown list.

File: packages/components/src/SingleSelect/examples/basic.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+        <br />
+        <br />
+        <br />
+        <br />
+        <br />
+        <br />
+        <br />
+        <br />
+        <br />
```
Comment (CWSites) [inline packages/components/src/SingleSelect/examples/basic.js]: Why are you having to force whitespace below the selects?

File: packages/components/src/defaultTheme/index.js [hunk 2]
```diff
@@ -66,11 +66,12 @@ const internalTheme = {
   Paper: {
     background: palette.paper,
     borderRadius: 4,
-    boxShadow: '0px 1px 5px 0px rgba(0, 0, 0, 0.2), 0px 2px 2px 0px rgba(0, 0, 0, 0.14), 0px 3px 1px -2px rgba(0, 0, 0, 0.12)',
+    boxShadow:
```
Comment (CWSites) [inline packages/components/src/defaultTheme/index.js]: for readability can you put each box shadow on a new line?

File: packages/components/src/defaultTheme/index.js [hunk 3]
```diff
@@ -66,11 +66,12 @@ const internalTheme = {
...
+      '0px 1px 5px 0px rgba(0, 0, 0, 0.2), 0px 2px 2px 0px rgba(0, 0, 0, 0.14), 0px 3px 1px -2px rgba(0, 0, 0, 0.12)',
     elevation: {
       boxShadow: '6px 6px 24px 0 rgba(30, 170, 189, 0.3)',
-      border: '1px solid #1EAABD'
-    }
+      border: '1px solid #1EAABD',
```
Comment (CWSites) [inline packages/components/src/defaultTheme/index.js]: There are a lot of colors throughout that should follow our existing palette but are hard coded instead?

File: packages/components/src/SingleSelect/examples/basic.js (hunk 1 above)
Comment (walkerrandolphsmith) [reply packages/components/src/SingleSelect/examples/basic.js]: Oh cool -- so when we used docz the "playground" was an iframe and therefore our calculation for dimensions thought we had less space then what was visually available. Looks like the new gatsby site doesn't have this consideration so we can remove them 🎊

**Comment by walkerrandolphsmith** [comment]
@jallers fixed
- Clearing the text in the input should stop highlighting the item in the pop-up list
- Alt+Down and Alt+Up should show and hide the dropdown list.

Did not fix
- filtering options when the list is rendered above the input renders the menu in the wrong location.

**Pull request merged**